packages/common/src/cloudwatch-event.ts

```diff
@@ -0,0 +1,88 @@
+import * as log from './log';
+import { isOneOf } from './util';
+import { AwsCloudWatchEvent } from './types';
+
+/**
+ * Determine if Cloudwatch event is a Step Function event
+ *
+ * @param {AwsCloudWatchEvent} event - A Cloudwatch event
+ * @returns {boolean} True if event is a Step Function event
+ */
+export const isSfExecutionEvent = (event: AwsCloudWatchEvent) =>
+  event.source === 'aws.states';
+
+/**
+ * Determine if Step Function is in a terminal state.
+ *
+ * @param {string} status - A Step Function execution status from a Cloudwatch event
+ * @returns {boolean} True if Step Function is in terminal state.
+ */
+export const isTerminalSfStatus: (status: string) => boolean = isOneOf([
+  'ABORTED',
+  'SUCCEEDED',
+  'FAILED',
+  'TIMED_OUT'
+]);
+
+/**
+ * Determine if Step Function is in a failed state.
+ *
+ * @param {Object} status - A Step Function execution status from a Cloudwatch event
+ * @returns {boolean} True if Step Function is in failed state.
+ */
+export const isFailedSfStatus: (status: string) => boolean = isOneOf([
+  'ABORTED',
+  'FAILED',
+  'TIMED_OUT'
+]);
+
+/**
+ * Get Step Function status from Cloudwatch event.
+ *
+ * @param {AwsCloudWatchEvent} event - A Cloudwatch event
+ * @returns {string|undefined} Step Function execution status
+ */
+export const getSfEventStatus = (event: AwsCloudWatchEvent) =>
+  event?.detail?.status;
```

Comment:
Oh my. So this will return `event.detail.status` if it exists, but also just return undefined if the `detail` or `detail.status` keys don't exist?

Reply:
Correct. [Optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) is coming in the next release of JS, and is currently supported in TS. It lets us get rid of `lodash.get`.

A related feature that you'll want to be aware of is [nullish coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/overview.html#nullish-coalescing)